### DISCUSSION TITLE
Can't able to add rtmp endpoint for h265 stream?

### DISCUSSION THREAD

**Discussion by cloud-tentosoft** [ant-media/Ant-Media-Server#6041] [General]
![image](https://github.com/ant-media/Ant-Media-Server/assets/155077398/dd02b7c8-c7e2-4ac6-94b6-6fa09ac84369)

![image](https://github.com/ant-media/Ant-Media-Server/assets/155077398/6a46bf80-8955-46cb-a7a3-3500009ab467)

**Comment by USAMAWIZARD** [comment]
@MuhammadYaqoobG Ant Media Does Not support H265 Over RTMP you can use SRT instead of RTMP if you are looking for streaming H265.
`ffmpeg -re -i ~/Videos/test.mp4 -c:v libx265 -c:a aac -f mpegts "srt://test.antmedia.io:4200?streamid=LiveApp/stream11"`

**Reply by cloud-tentosoft** [reply]
how to add in ant media dashboard, since rtmp endpoints only available?

**Reply by USAMAWIZARD** [reply]
Does your camera supports RTSP ??

**Comment by deadbeef84** [comment]
Now that ffmpeg supports h265 over RTMP, is there any chance antmedia could support it too?

**Reply by mekya** [reply]
Hi @deadbeef84,
Thank you for this question. 

Currently, it's not on the roadmap. We can put it into the roadmap if users, like you, are asking for this feature or if there is a good business opportunity.  

Cheers
Oguz

**Reply by deadbeef84** [reply]
We have a client that needs it for 4K HDR [restreaming to Youtube, which requires HEVC](https://support.google.com/youtube/answer/10265272?hl=en#zippy=%2Ccommon-encoder-configuration-requirements).

**Reply by mekya** [reply]
Hi @deadbeef84,

Thank you. I've [created an issue](https://github.com/ant-media/Ant-Media-Server/issues/6426) for the feasbility and it'll be scheduled probably this week because there is an increasing demand for this feature. 

FYI

Cheers
Oz

**Reply by deadbeef84** [reply]
That’s awesome! 👍

**Reply by mekya** [reply]
Hi @deadbeef84 ,

We've implemented the HEVC ingest in RTMP. It will be merged into the master in this week
https://github.com/ant-media/Ant-Media-Server/pull/6537

Pubishing H.265 to youtube requires us to upgrade FFmpeg to the latest version. It will be done through this issue 
https://github.com/ant-media/Ant-Media-Server/issues/6551 and already scheduled. 

Then you will directly push the stream to youtube with H.265(RTMP) ingested by H.265(RTMP)

Let me know if it's good for you or I have any misunderstanding. 

Cheers
Oz

**Reply by deadbeef84** [reply]
Sounds perfect, looking forward to the next release 👍

**Comment by amarantmeida** [comment]
@MuhammadYaqoobG Ant Media Server does not support H.265 over RTMP natively. Ant Media Server primarily supports H.264 (AVC) for RTMP streams, which aligns with the traditional and most widely used configuration for RTMP. To utilize H.265 with Ant Media Server, you might need to use these supported protocols (SRT, WebRTC, HLS/DASH) rather than RTMP.
